### ISSUE TITLE
zcl_mockup_loader_stub_factory->generate_stub( ) works only for interfaces with instance methods only

### ISSUE THREAD

**Issue by SBKubric** [sbcgua/mockup_loader#25]
Hello.
Thx for great work!

I've noticed that generated classes have the instance method `increment_count`. Therefore it is impossible to mock static methods in current realization. Is it bug or feature?

**Comment by sbcgua** [comment]
Hmmm. Difficult to say, because mocking of static methods was not intended at all. A typical scenario (we use in my team) is a data accessor. So all selects are in one class, and defined via interface - instance only methods. Unit tests create own local class, that implement same interface but connects it to the data with mockup loader instead of DB selects. Consequently methods are always instance methods. 

Can you share your case for static methods ? Also PR is welcomed if you see how to fix it (without breaking the main functionality) ;) 

P.S. But just intuitively I can say that static interface method is something that potentially need refactoring

**Comment by SBKubric** [comment]
Unfortunately, the main reason is bad architecture.
I am currently in process of refactoring legacy of the monolithic webapp, which wasn't supposed to use UT at all by design.
Still, the feature is crucial for such cases)

I've implemented it in my system, but i'm not sure if it is possible to make an PR -- the system has outdated SAP Basis so no AbapGit)
Long story short:
I digged a way to determine if the method is static (it is achieved with parsing result from FM SEO_METHODS_READ_ALL in factory class. It is not as tidy, as I wanted to, but works).
Also added a new static method to generated class - get_instance and a new static attribute - mo_me type ref to lcl_mockup_stub_dummy .

Then i divided logic in generate_mockup into two methods: generate_static_method and generate_instance_method .
generate_static_method copies generate_instance_method , except all method calls are made through local instance:
lo_me type ref to lcl_mockup_dummy .
lo_me = get_instance( ).
if lo_me->is_disabled( 'METHOD' ) = abap_true.
return.
endif.
...
mo_me is initialized during constructor 's call .
